### ISSUE TITLE
Find in expanded text lines only (ignoring all collapsed lines)

### ISSUE THREAD

**Issue by hakanedison** [microsoft/vscode#204712]
I often collapse regions of code that are not affected by my current task so that they are "out of the way" on the screen.

When searching a document (Ctrl + F) I want to be able to exclude all collapsed lines from the search, so that only text lines that are currently visible in the document are searched.

Otherwise, the search also finds text that is hidden, whereupon its code region is expanded, which is what I don't want.

Maybe you could add an extra button for this in the 'Find' box, with for example an eye icon?

FYI: I haven't considered whether it would also be beneficial to have the same functionality in the 'Search' panel (in multiple documents). I leave that to you to decide ;)

**Comment by gjsjohnmurray** [comment]
Previously requested in #168701

**Comment by hakanedison** [comment]
Yeah, but that one is closed. Maybe people have come to their senses now 😀.

My issue is not as much that I have to step through evey found instance, as it is that it expands my collapsed regions (when search term found inside).

(My suggestion is placing the button *inside* the Find box. Tooltip: "Skip collapsed lines".)

**Comment by hakanedison** [comment]
My suggestion visualized:

![Skip-collapsed-lines-(Find)](https://github.com/microsoft/vscode/assets/78951926/175b1379-3849-40dc-951c-c45f69c25371)
